web/src/main/java/org/bkvm/bookkeeper/restapi/HttpRequest.java

```diff
@@ -0,0 +1,128 @@
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.nio.client.CloseableHttpAsyncClient;
+import org.apache.http.impl.nio.client.HttpAsyncClients;
+import org.apache.http.util.EntityUtils;
+
+public class HttpRequest {
+
+    public static BookieApiResponse sendGetRequest(String httpServerUri, String path, boolean async) throws IOException, InterruptedException, URISyntaxException, ExecutionException {
+        BookieApiResponse gcResponse;
+        URI uri = buildUri(httpServerUri, path);
+        HttpGet request = new HttpGet(uri);
+        if (async) {
+            try (CloseableHttpAsyncClient httpclient = HttpAsyncClients.createDefault()) {
+                httpclient.start();
+                Future<HttpResponse> future = httpclient.execute(request, null);
+                HttpResponse response = future.get();
+                gcResponse = getResponse(response);
+            }
+        } else {
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    gcResponse = getResponse(response);
+                }
+            }
+        }
+        return gcResponse;
+    }
+
+    public static BookieApiResponse sendPostRequest(String httpServerUri, String path, Map<String, String> headers, boolean async) throws IOException, InterruptedException, URISyntaxException, ExecutionException {
+        BookieApiResponse gcResponse;
+        URI uri = buildUri(httpServerUri, path);
+        HttpPost request = new HttpPost(uri);
+        if (headers != null) {
+            headers.forEach((k, v) -> request.addHeader(k, v));
+        }
+        if (async) {
+            try (CloseableHttpAsyncClient httpclient = HttpAsyncClients.createDefault()) {
+                httpclient.start();
+                Future<HttpResponse> future = httpclient.execute(request, null);
+                HttpResponse response = future.get();
+                gcResponse = getResponse(response);
+            }
+        } else {
+            try (CloseableHttpClient httpClient = HttpClients.createDefault()) {
+                try (CloseableHttpResponse response = httpClient.execute(request)) {
+                    gcResponse = getResponse(response);
+                }
+            }
+        }
+        return gcResponse;
+    }
+
+    public static BookieApiResponse sendPutRequest(String httpServerUri, String path, boolean async, Map<String, String> headers) throws IOException, InterruptedException, URISyntaxException, ExecutionException {
+        BookieApiResponse gcResponse;
+        URI uri = buildUri(httpServerUri, path);
+        HttpPut request = new HttpPut(uri);
+        if (headers != null) {
+            headers.forEach((k, v) -> request.addHeader(k, v));
+        }
+        if (async) {
```

Comment:
I don't understand what is the purpose of this flag.

We are blocking and waiting for the response in any case

Reply:
Yes this is true